src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java

```diff
@@ -357,4 +361,45 @@ public void updateStage(UUID caseUUID, UUID stageUUID, Map<String, String> value
 
         camundaClient.completeTask(stageUUID, values);
     }
+
+    public ResponseEntity closeCase(UUID caseUUID) throws JsonProcessingException {
+        GetCaseworkCaseDataResponse caseDetails = caseworkClient.getCase(caseUUID);
+
+        //Get stage uuid from casework
+        String stage = caseworkClient.getActiveStage(caseDetails.getReference().replace("/", "%2F"));
+        UUID stageUUID = UUID.fromString(getStageUUID(stage));
+
+        //Mark case as complete
+        caseworkClient.completeCase(caseUUID, true);
+
+        //Update the stage team to null
+        UUID oldTeam = caseworkClient.getStageTeam(caseUUID, stageUUID); //To allow reversion if required
```

Comment:
Not a blocking comment, but should the 'completeCase' method in Casework->CaseService perform this action instead? you could put that behaviour in a transaction with the complete=true and not have to do the countermeasure.

Reply:
We'd need the Complete status and current team anyway in order to reset if killing the camunda process failed so I think this is still the way to go